libvirt.go

```diff
@@ -218,6 +243,49 @@ func (l *Libvirt) Domains() ([]Domain, error) {
 	return result.Domains, nil
 }
 
+// DomainState returns state of the domain managed by libvirt.
+func (l *Libvirt) DomainState(dom string) (DomainState, error) {
+	d, err := l.lookup(dom)
+	if err != nil {
+		return DomainStateNoState, err
+	}
+
+	req := struct {
+		Domain Domain
+		Flags  uint32
+	}{
+		Domain: *d,
+		Flags:  0,
+	}
+
+	buf, err := encode(&req)
+	if err != nil {
+		return DomainStateNoState, err
+	}
+
+	resp, err := l.request(constants.ProcDomainGetState, constants.ProgramRemote, &buf)
+	if err != nil {
+		return DomainStateNoState, err
+	}
+
+	r := <-resp
+	if r.Status != StatusOK {
+		return DomainStateNoState, decodeError(r.Payload)
+	}
+
+	result := struct {
```

Comment:
we're ignoring the `reason` value here. I'm okay with that for now but we should express the existence within the type:
```go
result := struct {
        State uint32
        Reason uint32
}
```